src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/Dashboard.test.tsx

```diff
@@ -3779,16 +3780,44 @@ describe("CSAT survey banner", () => {
     });
     await user.click(dismissCta);
 
-    // The petition CSAT survey is only shown on the next visit of the dashboard
-    // so that the user is not being flashed directly with a second banner after
-    // interacting with the petition banner.
-    render(<ComposedDashboardComponent />);
-
     const answerButton = screen.getByRole("button", {
       name: "Neutral",
     });
     expect(answerButton).toBeInTheDocument();
   });
+
+  it("displays the petition CSAT survey for users in the treatment branch after they clicked “Sign petition”", async () => {
+    const user = userEvent.setup();
+    const ComposedDashboard = composeStory(
+      DashboardUsPremiumResolvedScanNoBreaches,
+      Meta,
+    );
+    const ComposedDashboardComponent = () => (
+      <ComposedDashboard
+        activeTab="fixed"
+        enabledFeatureFlags={["PetitionBannerCsatSurvey"]}
+        experimentData={{
+          ...defaultExperimentData,
+          "data-privacy-petition-banner": {
+            enabled: true,
+          },
+        }}
+      />
+    );
+    render(<ComposedDashboardComponent />);
+
+    const dismissCta = screen.getByRole("link", {
+      name: "Sign petition",
+    });
+    await user.click(dismissCta);
+
+    await waitFor(() => {
```

Comment:
This `waitFor` surprised me - maybe deserves a comment about why it's necessary?

Reply:
Good callout — I see how the use of `waitFor` is unexpected here at first. I’ve added a comment in [caf277b](https://github.com/mozilla/blurts-server/pull/4856/commits/caf277b2f5170338723f2c169f2d2f9cf2d8d42a).